asn.go

```diff
@@ -56,6 +52,17 @@ func (a *asnStore) AsnForIPv6(ip net.IP) (string, error) {
 // NewAsnStore returns a `asnStore` that can be queried for the Autonomous System Numbers
 // for a given IP address or a multiaddress which contains an IP address.
 func NewAsnStore() (*asnStore, error) {
+	if Store.store == nil {
```

Comment:
race condition. you must check the channel.

Reply:
It's not clear to me why this function even exists/is exported. Scrap it?

Reply:
yes, better to scrap it.